sklearn/datasets/openml.py

```diff
@@ -0,0 +1,306 @@
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+        else:
+            raise error
+
+    if not data_found and version != "active":
+        # might have been deactivated. will warn later
+        data_found = True
+        try:
+            url = (_SEARCH_NAME +
+                   "/data_version/{}/status/deactivated").format(name, version)
+            response = urlopen(url)
+        except HTTPError as error:
+            # 412 is an OpenML specific error code, indicating a generic error
+            # (e.g., data not found)
+            if error.code == 412:
+                data_found = False
+            else:
+                raise error
+
+    if not data_found:
+        # not in except for nicer traceback
+        if version == "active":
+            raise ValueError("No active dataset {} found.".format(name))
+        raise ValueError("Dataset {} with version {}"
+                         " not found.".format(name, version))
+
+    json_data = json.loads(response.read().decode("utf-8"))
+    return json_data['data']['dataset'][0]
+
+
+def _get_data_description_by_id(data_id):
+    data_found = True
+    try:
+        response = urlopen(_DATA_INFO.format(data_id))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+    if not data_found:
+        # not in except for nicer traceback
+        raise ValueError("Dataset with id {} "
+                         "not found.".format(data_id))
+    json_data = json.loads(response.read().decode("utf-8"))
+    response.close()
+    return json_data['data_set_description']
+
+
+def _get_data_features(data_id):
+    data_found = True
+    try:
+        response = urlopen(_DATA_FEATURES.format(data_id))
+    except HTTPError as error:
+        # 412 is an OpenML specific error code, indicating a generic error
+        # (e.g., data not found)
+        if error.code == 412:
+            data_found = False
+        else:
+            raise error
+    if not data_found:
+        # not in except for nicer traceback
+        raise ValueError("Dataset with id {} "
+                         "not found.".format(data_id))
+    json_data = json.loads(response.read().decode("utf-8"))
+    response.close()
+    return json_data['data_features']['feature']
+
+
+def _download_data(file_id):
+    url = _openml_fileid_url(file_id)
+    response = urlopen(url)
+    if sys.version_info[0] == 2:
+        # Python2.7 numpy can't handle unicode?
+        arff = loads(response.read())
+    else:
+        arff = loads(response.read().decode('utf-8'))
+
+    response.close()
+    return arff
+
+
+def _convert_numericals(data, name_feature):
+    for feature in name_feature.values():
+        if feature['data_type'] == "numeric":
+            idx = int(feature['index'])
+            data[:, idx] = data[:, idx].astype(np.float)
+    return data
+
+
+def fetch_openml(id=None, name=None, version='active', data_home=None,
```

Comment:
We could also switch the order of `id` and `name`? (so you can still do `fetch_openml('miceprotein', version=4)` instead of needing  `fetch_openml(name='miceprotein', version=4)`

Reply:
But given my comments below, maybe let's first fully decide on whether to combine both in a single parameter or not